features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourceDataStoreManagerTest.kt

```diff
@@ -0,0 +1,411 @@
+        )
+
+        // When
+        testedDataStoreManager.cacheResourceHash(fakeHash)
+
+        // Then
+        verify(mockFeatureScope.dataStore).setValue(
+            key = eq(DATASTORE_HASHES_ENTRY_NAME),
+            data = any(),
+            version = anyOrNull(),
+            callback = any(),
+            serializer = eq(mockResourceHashesEntrySerializer)
+        )
+    }
+
+    @Test
+    fun `M be threadsafe W cacheResourceHash() { store on one thread, read on another }`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+        val writeThread = Runnable {
+            testedDataStoreManager.cacheResourceHash(fakeHash)
+        }
+
+        val readThread = Runnable {
+            assertThat(testedDataStoreManager.isPreviouslySentResource(fakeHash))
+                .isTrue()
+        }
+
+        // When
+        writeThread.run()
+
+        // Then
+        readThread.run()
+    }
+
+    // region init
+
+    @Test
+    fun `M query entry from dataStore W init()`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+
+        // Then
+        verify(mockDataStoreHandler).value(
+            key = eq(DATASTORE_HASHES_ENTRY_NAME),
+            version = anyOrNull(),
+            callback = any<DataStoreReadCallback<ResourceHashesEntry>>(),
+            deserializer = any()
+        )
+    }
+
+    @Test
+    fun `M overwrite last update date W init() { datastore expired }`(
+        @Mock mockDataStoreContentEntry: DataStoreContent<ResourceHashesEntry>,
+        @Mock mockResourceHashesEntry: ResourceHashesEntry,
+        @IntForgery fakeVersionCode: Int
+    ) {
+        // Given
+        val fakeHashes = DataStoreContent(
+            versionCode = fakeVersionCode,
+            data = mockResourceHashesEntry
+        )
+
+        whenever(mockDataStoreContentEntry.versionCode).thenReturn(fakeVersionCode)
+        val expiredTimestamp = System.nanoTime() - (DATASTORE_EXPIRATION_NS * 2)
+        whenever(mockResourceHashesEntry.lastUpdateDateNs).thenReturn(expiredTimestamp)
+
+        whenever(
+            mockDataStoreHandler.value(
+                key = eq(DATASTORE_HASHES_ENTRY_NAME),
+                version = anyOrNull(),
+                callback = any<DataStoreReadCallback<ResourceHashesEntry>>(),
+                deserializer = any()
+            )
+        ) doAnswer {
+            @Suppress("UNCHECKED_CAST")
+            val callback = it.arguments[2] as DataStoreReadCallback<ResourceHashesEntry>
```

Comment:
variant without `Suppress` (same can be applied to the similar lines below)

```suggestion
            val callback = it.getArgument<DataStoreReadCallback<ResourceHashesEntry>>(2)
```